src/middleware/messageStore.coffee

```diff
@@ -65,70 +69,119 @@ exports.storeTransaction = (ctx, done) ->
     else
       ctx.transactionId = tx._id
       ctx.header['X-OpenHIM-TransactionID'] = tx._id.toString()
+      ctx.header['channel-id'] = tx.channelID
       return done null, tx
 
 exports.storeResponse = (ctx, done) ->
   logger.info 'Storing response for transaction: ' + ctx.transactionId
+  getNumRoutes ctx.authorisedChannel._id, (numRoutes) ->
+    status = null
+    if (500 <= ctx.response.status <= 599)
+      status = transactionStatus.FAILED
+    else
+      if (200 <= ctx.response.status <= 299)
+        console.log 'in Primary successful'
+        status = transactionStatus.SUCCESSFUL
+
+      # In all other cases mark as completed
+      if status is null or status is undefined
+        console.log 'in Primary Completed'
+        status = transactionStatus.COMPLETED
+
+    headers = copyMapWithEscapedReservedCharacters ctx.response.header
+
+    res =
+      status: ctx.response.status
+      headers: headers
+      body: if not ctx.response.body then "" else ctx.response.body.toString()
+      timestamp: ctx.response.timestamp
+
+
+    # check if channel response body is false and remove
+    if ctx.authorisedChannel.responseBody == false
+      # reset request body - primary route
+      res.body = ''
+
+      # reset request body - routes
+      if ctx.routes
```

Comment:
We don't need to do this route checking if we aren't storing route responses here. Just make sure this logic moves to where we are storing route responses.